File: Android开发资料/网关和云平台双模式轮询数据工具类示例/MainActivity.java
package org.shanguanling.b;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    /**
     * @author: ShanguanLinG
     * @since 2026/01/29 /12:50
     */

    CloudClient cloudClient;
    GateWayClient gateWayClient;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
        startCloudClient();
    }

    private void startCloudClient() {
        cloudClient = new CloudClient("account", "password");
        cloudClient.connect().thenRun(() -> cloudClient.startPolling(1322935));
        cloudClient.setListener(data -> System.out.println(data.get("l_co")));
        gateWayClient = new GateWayClient("172.18.23.16", 57500);
        gateWayClient.connect().thenRun(() -> gateWayClient.startPoller(new String[]{"uhf"}));
        gateWayClient.setListener(data -> System.out.println(data.get("uhf")));
    }

    private void initViews() {
    }
}

File: Android开发资料/Kotlin版的云平台和网关api调用示例/MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    /**
     * @author ShanguanLinG
     * @since 2026/01/12
     * nle_mylibrary包控制网关设备与获取网关设备数据示例
     *
     *
     * 需要保证平板的IP配置正确，且AndroidManifest.xml中写了
     * <uses-permission android:name="android.permission.INTERNET"></uses-permission>
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Loader.load()
    }
}